xdsl/interpreters/ml_program.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any
+
+from xdsl.dialects import ml_program
+from xdsl.dialects.builtin import DenseIntOrFPElementsAttr, TensorType
+from xdsl.interpreter import (
+    Interpreter,
+    InterpreterFunctions,
+    impl,
+    register_impls,
+)
+from xdsl.interpreters.shaped_array import ShapedArray
+
+
+@register_impls
+class MLProgramFunctions(InterpreterFunctions):
+
+    @impl(ml_program.Global)
+    def run_global(
+        self, interpreter: Interpreter, op: ml_program.Global, args: tuple[Any, ...]
+    ) -> tuple[Any, ...]:
+        if op.is_mutable is not None:
+            raise NotImplementedError(
+                "mutable global not yet supported in ml_program.global interpreter"
+            )
+        global_value = op.value
+        assert isinstance(global_value, DenseIntOrFPElementsAttr)
+        shape = global_value.get_shape()
```

Comment:
I think better raise not implemented if shape is none here, I'm not sure when this would arise but it feels like we want to catch this explicitly